File: wlr/SPX.Sprites/SpriteSheetManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;

namespace SPX.Sprites
{
    public class SpriteDefinition
    {
        public SpriteDefinition(int type, int index, int frames)
        {
            Type = type;
            Info = new SpriteInfo(index, frames);
        }
        public int Type { get; set; }
        public SpriteInfo Info { get; set; }
    }
    public class ISpriteInitializer
    {
        public virtual ICollection<SpriteDefinition> GetSprites() { return null; }
    }
    public class SpriteSheetManager
    {
        private static Dictionary<int, SpriteInfo> __manager = new Dictionary<int, SpriteInfo>();

        public static SpriteInfo GetSpriteInfo(int spriteName)
        {
            return __manager[spriteName];
        }

        public static void Setup(ISpriteInitializer initializer)
        {
            foreach (var sprite in initializer.GetSprites())
            {
                __manager.Add(sprite.Type, sprite.Info);
            }
        }
    }
}

File: code/spx-ported/src/com/spx/sprites/SpriteSheetManager.java
package com.spx.sprites;import com.xna.wrapper.*;import java.util.*;
    public class SpriteSheetManager
    {
        private static HashMap<Integer, SpriteInfo> __manager = new HashMap<Integer, SpriteInfo>();

        public static SpriteInfo GetSpriteInfo(int spriteName)
        {
            return __manager.get(spriteName);
        }

        public static void Setup(ISpriteInitializer initializer)
        {
            for(SpriteDefinition sprite:initializer.GetSprites())
            {
                __manager.put(sprite.Type, sprite.Info);
            }
        }
    }
